Upgrade to Jetty 12

jcoyne opened the pull request.

jcoyne committed:
Upgrade to Jetty 12

DiegoPino committed:
Humble attempt to make Jetty 12 happy/ static ::from() to generate Constraint

No server tests are passing but it compiles. The Changes in/methods for classes like server.Request have changed. Probably need to be all refactored and then we need to actually test if this works or not?

@jcoyne please feel free to use as an idea/revert/refactor at will

zzmloveddm commented:
> _ No description provided. _
Hello, I pulled your branch and it prompted 404 after startup. Why is this?

DiegoPino commented:
@zzmloveddm this is a WIP branch and also marked as DRAFT. All tests need to be rewritten and we have not even made sure what we have works. That applies to every branch that is not development (where what works gets merged) or the actual release branches. If you want to contribute to the jetty v12 upgrade efforts please feel free to fork/work on a local branch and make a commit. Thanks

zzmloveddm commented:
> @zzmloveddm this is a WIP branch and also marked as DRAFT. All tests need to be rewritten and we have not even made sure what we have works. That applies to every branch that is not development (where what works gets merged) or the actual release branches. If you want to contribute to the jetty v12 upgrade efforts please feel free to fork/work on a local branch and make a commit. Thanks

Thanks

DiegoPino commented on src/main/java/edu/illinois/library/cantaloupe/http/Server.java:
```diff
@@ -71,118 +66,93 @@ public Server() {
         } else {
-            server.setHandler(handler);
+            // Set the handler directly if no authentication
+            ContextHandler contextHandler = new ContextHandler("/");
```
I am not totally sure we need the ContextHandler here .. this might be why a user was getting a 404 when testing this branch out.

zzmloveddm replied:
https://github.com/jetty/jetty.project/issues/11378

zzmloveddm committed:
Fix the issue of 404 after upgrading Jetty12

glenrobson committed:
Upgrade Kakadu to 8.4.1

glenrobson committed:
Testing windows

glenrobson committed:
Installing libjpeg-turbo

glenrobson committed:
Removing tests and created https://github.com/cantaloupe-project/cantaloupe/issues/736

DiegoPino committed:
Adds exception for Video Resources where info JSON will only contain

Size (and image info) for the first frame bc there is not such thing as variable frame size Video and also a Video could possible contain 60fps by the length images, which would basically lead to a info.json of HUGE sizes.

By tapping into the getSize method and adding a helper function (isVideo) to the Info Class we avoid rewriting every v1/v2/v3 request for getSize(index) where a video could be involved. This seems cleaner

DiegoPino committed:
Adds a simple test for the new isVideo() Method.

Makes sense to me to make it here so i can reuse the Video, itself. Will add another at the info Test class too

DiegoPino committed:
Adds isVideo and not Video tests

DiegoPino committed:
make tests happy

glenrobson committed:
Adding readme showing where libs should go

glenrobson committed:
Updating security issues versions

glenrobson committed:
Regenerating Proto java file

DiegoPino committed:
Merge branch 'develop' into jetty-12

DiegoPino committed:
Merge branch 'zzmloveddm-jetty-12-fix-404' into jetty-12

DiegoPino committed:
kHttpHTTPImageInputStreamClientTest: Rewrite overrides of DefaultHandler() for Jetty 12

DiegoPino committed:
ServerTest: rewrite DefaultHandler() override for Jetty 12

DiegoPino committed:
HttpSourceTest: rewrite DefaultHandler() for Jetty 12

DiegoPino committed:
Jakarta 6 (new stuff, more abstract methods/others gone): this now compiles, but i am getting errors on testing

(maybe transient?)

jcoyne commented:
Fixed by a newer PR
